src/services/types.ts

```diff
@@ -182,6 +182,10 @@ namespace ts {
         isCancellationRequested(): boolean;
     }
 
+    export interface CancellationToken {
+        throwIfCancellationRequested(cleanup?: () => void): void;

```

Comment:
I added this instead of just asking `isCancellationRequested` and throwing on my own because I wanted to keep the built-in tracing.